src/som/vmobjects/SInteger.java

```diff
@@ -24,19 +25,38 @@
 
 package som.vmobjects;
 
-import java.math.BigInteger;
-
 import som.vm.Universe;
 
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.Map;
+
 public final class SInteger extends SNumber {
 
+  /**
+   * Language convention requires integers up to this value to be identical.
+   */
+  private static final long MAX_IDENTICAL_INT = 1073741823L;
```

Comment:
For SOM, and CSOM, no. I didn't really touch them. So, simple is good. Only TruffleSOM and RTruffleSOM optimize for performance.

About the test, I read it a little different. The `42 == 42` (`assert: a == b`) seems the only hard requirement, the rest are soft requirements.

I am mostly asking, because I would still like to be able to run the benchmarks, and this seems likely a major performance issue.
Perhaps a Java-like solution with an array for the range of something like -5 to 250 or so would be sufficient. A `HashMap` for 1073741823 values seems a bit much :)
But I am happy that the tests pass. I might come back to this if I really need it.